virtual_rainforest/soil/model.py

```diff
@@ -0,0 +1,67 @@
+"""The `soil.model` module.
+
+The `soil.model` module creates a `SoilModel` class, which extended the base `Model`
+class to be usable to simulate the soil.
+"""
+
+from numpy import datetime64, timedelta64
+
+from virtual_rainforest.core.logger import LOGGER, log_and_raise
+from virtual_rainforest.core.model import BaseModel
+
+
+class InitialisationError(Exception):
+    """Custom exception class for model initialisation failures."""
+
+    pass
+
+
+class SoilModel(BaseModel):
+    """A class describing the soil model.
+
+    Describes the specific functions and attributes that the soil module should possess.
+    It's very much incomplete at the moment, and only overwrites one function in a
+    pretty basic manner. This is intended as a demonstration of how the class
+    inheritance should be handled for the model classes.
+
+    Args:
+        start_time: Point in time that the model simulation should be started.
+        end_time: Time that the model simulation should end
+        update_interval: Time to wait between updates of the model state.
+
+    Attributes:
+        name: Names the model that is described
+    """
+
+    name = "soil model"
+
+    # THIS IS BASICALLY JUST A PLACEHOLDER TO DEMONSTRATE HOW THE FUNCTION OVERWRITING
+    # SHOULD WORK
+    def setup(self) -> None:
+        """Function to set up the soil model."""
+        for layer in range(0, self.no_layers):
+            LOGGER.info(f"Setting up soil layer {layer}")
+
+    def __repr__(self) -> str:
+        """Represent a Model as a string."""
+        return (
+            f"SoilModel(start_time={self.start_time}, end_time={self.end_time}, "
```

Comment:
See above

Reply:
This definitely saves a lot of effort in terms of not having to write a new `__repr__` for every new model instance. I was wondering though whether there's anyway to generate to the function signature automatically, as this changes for the inheriting classes? Or alternatively if using the `__repr__` to print the full function signature is not actually a sensible thing to do?

Reply:
I've never used it myself, but probably you could see if you can exploit `inspect.Signature` for this: https://docs.python.org/3/library/inspect.html#inspect.Signature